src/main/kotlin/uk/gov/gdx/datashare/models/CreateSupplierRequest.kt

```diff
@@ -1,10 +1,23 @@
 package uk.gov.gdx.datashare.models
 
 import io.swagger.v3.oas.annotations.media.Schema
+import org.hibernate.validator.constraints.Length
 import uk.gov.gdx.datashare.enums.EventType
+import javax.validation.constraints.Pattern
 
 data class CreateSupplierRequest(
-  @Schema(description = "Name of client, must be lower case", required = true, example = "HMPO")
+  @Schema(
+    description = "Name of client, may only contain lowercase letters, numbers, and the following special characters: + = , . @ -",
+    required = true,
+    example = "hmpo",
+    maxLength = 128,
+    pattern = "^[a-z0-9+=,.@-]*\$",
+  )
+  @get:Length(min = 3, max = 128)
+  @get:Pattern(
+    regexp = "^[a-z0-9+=,.@-]*\$",
+    message = "Name may only contain lowercase letters, numbers, and the following special characters: + = , . @ -",
+  )
   val clientName: String,
```

Comment:
That we've written the same set of constraints twice, is I think the strongest reason I could come up with for pulling this into a type. I'm not too worried here, but I think it's worth keeping in mind.